client/store/actions/file.ts

```diff
@@ -163,19 +165,31 @@ export async function locateFile(path: string) {
   })
 }
 
-export async function selectFile(newPath?: string) {
+export async function selectFile(newPath?: string[] | string) {
   const { path, record } = store.getState()
-  if (path === newPath) return
+  if (newPath) {
+    if (!Array.isArray(newPath)) newPath = [newPath]
+    if (newPath.length === 1) {
+      store.setState('select-file-reset', (state) => {
+        state.selectedMultiplePaths = undefined
+      })
 
-  store.setState('select-file', (state) => {
-    state.path = newPath
-  })
+      if (path === newPath[0]) return
+      store.setState('select-file-start', (state) => {
+        state.path = newPath?.[0]
+      })
 
-  if (!newPath) return
-  if (record?.path === newPath) return
-  if (getIsFolder(store.getState())) return
+      if (!newPath) return
+      if (record?.path === newPath[0]) return
+      if (getIsFolder(store.getState())) return
 
-  await openFile(newPath)
+      await openFile(newPath[0])
+    } else {
+      store.setState('select-file-end', (state) => {
+        state.selectedMultiplePaths = newPath as string[]
```

Comment:
@roll i agree with this observation, however, I added this to mitigate the side effects of introducing this change, because the data does get passed as an array to this file, but since `selectFile` is used by other parts of the store, like [adjustFile](https://github.com/okfn/opendataeditor/blob/main/client/store/actions/file.ts#L95), [onFileCreate](https://github.com/okfn/opendataeditor/blob/main/client/store/actions/event.ts#L55) where it was originally expected as a string, this is necessary. My suggestion is to create a cleanup PR separate from this one to deal with this issue after we merge this one.

Reply:
@guergana 
Yeah, sure. It's just a suggestion. We, for example, can add a TODO to this line.